src/WebUI/WorkLoads/WorkloadDetails.tsx

```diff
@@ -61,29 +68,75 @@ export interface IWorkLoadDetailsItem {
 export class WorkloadDetails extends BaseComponent<IWorkloadDetailsProperties, IWorkloadDetailsState> {
     constructor(props: IWorkloadDetailsProperties) {
         super(props, {});
+        this._podsStore = StoreManager.GetStore<PodsStore>(PodsStore);
+
+        this._podsStore.addListener(PodsEvents.PodsFetchedEvent, this._onPodsUpdated);
+
+        this._workloadsStore = StoreManager.GetStore<WorkloadsStore>(WorkloadsStore);
+
+        let item = props.item;
+        if (!props.item) {
+            const kubeService = KubeSummary.getKubeService();
+            ActionsCreatorManager.GetActionCreator<PodsActionsCreator>(PodsActionsCreator).getPods(kubeService);
```

Comment:
We already fetch all the pods in WorkloadsPivot..and they are stored in PodsStore. Do we need to fetchhere again

Reply:
If we land directly on the workload details, we won't be going through WorkloadPivot (that is only if we land on the landing view